pkgs/objective_c/lib/src/internal.dart

```diff
@@ -76,70 +84,123 @@ final msgSendStretPointer =
 final useMsgSendVariants =
     Abi.current() == Abi.iosX64 || Abi.current() == Abi.macosX64;
 
-/// Only for use by ffigen bindings.
-class _ObjCFinalizable<T extends NativeType> implements Finalizable {
-  final Pointer<T> _ptr;
-  bool _pendingRelease;
+// _FinalizablePointer exists because we can't access `this` in the initializers
+// of _ObjCReference's constructor, and we have to have an owner to attach the
```

Comment:
You could try to use factories instead.

But that would make it diverge from from the JNI implementation. @HosseinYousefi Did we consider if `_JFinalizable` could be merged with `JGlobalReference` when marking the latter one `Finalizable`?

Reply:
Ah of course! 😄  hash-tag-I-implemented-deeply-immutable 😆

Reply:
The other issue in my case is `_ObjCReference` has 2 subclasses, `_ObjCObjectRef` and `_ObjCBlockRef`, which need to call `_ObjCReference`'s constructor in their constructor. But you can't invoke a factory constructor like this (since there's no guarantee it returns a fresh instance, or even that it returns an instance of that type).